vision/src/autogluon/vision/_gluoncv/image_classification.py

```diff
@@ -0,0 +1,453 @@
+"""Auto pipeline for image classification task"""
+# pylint: disable=bad-whitespace,missing-class-docstring,bare-except
+import time
+import os
+import math
+import copy
+import logging
+import pprint
+import json
+import pickle
+from typing import Union, Tuple
+import uuid
+import shutil
+
+import numpy as np
+import pandas as pd
+import autogluon.core as ag
+from autocfg import dataclass
+from autogluon.core.constants import BINARY, MULTICLASS, REGRESSION
+from autogluon.core.decorator import sample_config
+from autogluon.core.scheduler.resource import get_cpu_count, get_gpu_count
+from autogluon.core.task.base import BaseTask
+from autogluon.core.searcher import RandomSearcher
+
+from gluoncv.auto.estimators.base_estimator import BaseEstimator
+from gluoncv.auto.estimators import ImageClassificationEstimator, TorchImageClassificationEstimator
+from gluoncv.auto.data.dataset import ImageClassificationDataset
+from gluoncv.auto.estimators.conf import _BEST_CHECKPOINT_FILE
+
+from .utils import config_to_nested
+
+
+__all__ = ['ImageClassification', 'ImagePrediction']
+
+try:
+    import timm
+except ImportError:
+    timm = None
+try:
+    import torch
+except ImportError:
+    torch = None
+try:
+    import mxnet as mx
+    from gluoncv.model_zoo import get_model_list
+except ImportError:
+    mx = None
```

Comment:
This logic is a bit concerning if we have to load mxnet and torch upfront rather than lazily.

Don't need to address in this PR, but something to think about for later refactoring.